src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseTestCase.java

```diff
@@ -78,12 +81,19 @@ public String getTestName() {
     }
 
     public ProgrammingExerciseTestCase testName(String testName) {
-        this.testName = testName;
+        this.testName = truncateTestName(testName);
         return this;
     }
 
     public void setTestName(String testName) {
-        this.testName = testName;
+        this.testName = truncateTestName(testName);
+    }
+
+    private String truncateTestName(String testName) {
+        if (testName != null && testName.length() > MAX_TEST_CASE_LENGTH) {
+            return testName.substring(0, MAX_TEST_CASE_LENGTH);
+        }
+        return testName;
```

Comment:
```suggestion
        return StringUtils.truncate(testName, MAX_TEST_CASE_LENGTH);
```
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#truncate-java.lang.String-int-